src/openrct2/object/ObjectRepository.cpp

```diff
@@ -95,17 +78,129 @@ using ObjectEntryMap = std::unordered_map<rct_object_entry, size_t, ObjectEntryH
+            object->SetRepositoryItem(&item);
+            delete object;
+            return std::make_tuple(true, item);
+        }
+        else
+        {
+            return std::make_tuple(true, ObjectRepositoryItem());
+        }
+    }
+
+protected:
+    void Serialise(IStream * stream, const ObjectRepositoryItem &item) const override
+    {
+        stream->WriteValue(item.ObjectEntry);
+        stream->WriteString(item.Path);
+        stream->WriteString(item.Name);
+
+        switch (item.ObjectEntry.flags & 0x0F) {
+        case OBJECT_TYPE_RIDE:
+            stream->WriteValue<uint8>(item.RideFlags);
+            for (sint32 i = 0; i < 2; i++)
+            {
+                stream->WriteValue<uint8>(item.RideCategory[i]);
+            }
+            for (sint32 i = 0; i < MAX_RIDE_TYPES_PER_RIDE_ENTRY; i++)
+            {
+                stream->WriteValue<uint8>(item.RideType[i]);
+            }
+            stream->WriteValue<uint8>(item.RideGroupIndex);
+            break;
+        case OBJECT_TYPE_SCENERY_SETS:
+            stream->WriteValue<uint16>(item.NumThemeObjects);
+            for (uint16 i = 0; i < item.NumThemeObjects; i++)
+            {
+                stream->WriteValue<rct_object_entry>(item.ThemeObjects[i]);
+            }
+            break;
+        }
+    }
+
+    ObjectRepositoryItem Deserialise(IStream * stream) const override
+    {
+        ObjectRepositoryItem item = { 0 };
+
+        item.ObjectEntry = stream->ReadValue<rct_object_entry>();
+        item.Path = stream->ReadString();
+        item.Name = stream->ReadString();
+
+        switch (item.ObjectEntry.flags & 0x0F) {
+        case OBJECT_TYPE_RIDE:
+            item.RideFlags = stream->ReadValue<uint8>();
+            for (sint32 i = 0; i < 2; i++)
+            {
+                item.RideCategory[i] = stream->ReadValue<uint8>();
+            }
+            for (sint32 i = 0; i < MAX_RIDE_TYPES_PER_RIDE_ENTRY; i++)
+            {
+                item.RideType[i] = stream->ReadValue<uint8>();
+            }
+            item.RideGroupIndex = stream->ReadValue<uint8>();
+            break;
+        case OBJECT_TYPE_SCENERY_SETS:
+            item.NumThemeObjects = stream->ReadValue<uint16>();
+            item.ThemeObjects = Memory::AllocateArray<rct_object_entry>(item.NumThemeObjects);
+            for (uint16 i = 0; i < item.NumThemeObjects; i++)
+            {
+                item.ThemeObjects[i] = stream->ReadValue<rct_object_entry>();
+            }
+            break;
+        }
+        return item;
+    }
+
+private:
+    bool IsTrackReadOnly(const std::string &path) const
+    {
+        return
+            String::StartsWith(path, SearchPaths[0]) ||
+            String::StartsWith(path, SearchPaths[1]);
+    }
+};
+
 class ObjectRepository final : public IObjectRepository
 {
-    const IPlatformEnvironment *        _env = nullptr;
+    IPlatformEnvironment * const        _env = nullptr;
```

Comment:
This changes meaning of the `const` from "pointer to const" to "const pointer", please revert or change to `IPlatformEnvironment const *`

Reply:
I think this was what was originally intended. I mean for these to be immutable references, not read-only access - although I suppose both would be okay - but they need to be changed everywhere.